wallet/src/main/kotlin/net/thechance/wallet/service/AdminWalletService.kt

```diff
@@ -0,0 +1,32 @@
+package net.thechance.wallet.service
+
+import net.thechance.wallet.api.dto.balance.DepositRequest
+import net.thechance.wallet.api.dto.balance.toTransaction
+import net.thechance.wallet.exception.BlockedWalletUserException
+import net.thechance.wallet.repository.TransactionRepository
+import net.thechance.wallet.repository.WalletUserRepository
+import java.util.*
+
+class AdminWalletService(
+    private val transactionRepository: TransactionRepository,
+    private val walletUserRepository: WalletUserRepository,
+    private val blockService: BlockService
+) {
+
+    fun deposit(userId: UUID, request: DepositRequest) {
```

Comment:
A function in the service shouldn't depend on a request in the api layer you can the pass the parameters explicitly

```suggestion
    fun deposit(adminId: UUID, receiverPhoneNumber: String, amount: Double) {
```